src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -192,12 +194,13 @@ internal override bool IsExpressionBodied
             MethodKind methodKind,
             bool isAutoPropertyAccessor,
             DiagnosticBag diagnostics) :
-            base(containingType, syntax.GetReference(), syntax.Body?.GetReference(), location)
+            base(containingType, syntax.GetReference(), syntax.Body?.GetReference() ?? syntax.ExpressionBody?.GetReference(), location)
         {
             _property = property;
             _explicitInterfaceImplementations = explicitInterfaceImplementations;
             _name = name;
             _isAutoPropertyAccessor = isAutoPropertyAccessor;
+            _isExpressionBodied = (syntax.Body == null && syntax.ExpressionBody != null) ? true : _property.IsExpressionBodied;
```

Comment:
See edited comment above. It's better, I think.
I try to understand, why `_property.IsExpressionBodied` was used for `this.IsExpressionBodied` in the first place? Guess, because if it had *any* accessor it *cannot* have been expression bodied. But even that logic is a bit hidden?!